frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -436,3 +425,21 @@ export default _.compose(
   })),
   titleWithLoadingTime("queryStartTime"),
 )(QueryBuilder);
+
+function useShouldShowUnsavedChangesWarning() {
```

Comment:
Nice catch. I didn't realize that.

Reply:
On second thought, moving this hook to `frontend/src/metabase/query_builder/selectors.js` would require me to add unit tests for the new selector to ensure that it would work when called elsewhere. However, this feels to me that the logic only belongs to `QueryBuilder.jsx` alone, not any other component under `metabase/query_builder/*`. So leaving this here has some benefits compared to moving it to the selector file:
1. I don't need to create new tests which could be cumbersome to set up.
2. The logic is colocated. Since it's only being used in this file alone.

With the above 2 benefits, what do you think about leaving this as is?

Reply:
No, I think it's incorrect to combine selectors with hooks - you should add a selector instead. Feel free to test this change in `QueryBuilder` as you do now